src/EIP7702Proxy.sol

```diff
@@ -144,4 +161,71 @@ contract EIP7702Proxy is Proxy {
     function _implementation() internal view override returns (address) {
         return ERC1967Utils.getImplementation();
     }
+
+    receive() external payable {}
+
+    /**
+     * @notice Resets the ERC-1967 implementation slot after signature verification from any valid owner
+     * @dev Uses the implementation's isValidSignature to verify owner signatures
+     * @param newImplementation The implementation address to set
+     * @param nonce The nonce for this operation
+     * @param signature The SignatureWrapper from any valid owner
+     */
+    function resetImplementation(
+        address newImplementation,
+        uint256 nonce,
+        bytes calldata signature
+    ) external {
+        // Verify nonce hasn't been used
+        if (
+            !INonceTracker(nonceTracker).verifyAndUseNonce(address(this), nonce)
+        ) {
+            revert NonceAlreadyUsed();
+        }
+
+        // Hash includes newImplementation and nonce
+        bytes32 hash = keccak256(abi.encode(newImplementation, nonce));
+
+        // First try validating against new implementation's isValidSignature
+        (bool success, bytes memory result) = newImplementation.call(
+            abi.encodeWithSelector(
+                IERC1271.isValidSignature.selector,
+                hash,
+                signature
+            )
+        );
```

Comment:
this seems like it could be a critical because someone who write a `newImplementation` to validate with a different key?

Reply:
I think this style could work if we did the dedicated recovery contract path that was bound to a specific implementation address, for our case smart wallet v1. It would assume that previous storage existed for owners, truly making this a recovery-only scenario by leveraging existing state.